etna/models/seasonal_ma.py

```diff
@@ -28,165 +26,155 @@ def __init__(self, window: int = 5, seasonality: int = 7):
         self.window = window
         self.seasonality = seasonality
-        self.shift = self.window * self.seasonality
 
-    def fit(self, df: pd.DataFrame, regressors: List[str]) -> "_SeasonalMovingAverageModel":
-        """
-        Fit SeasonalMovingAverage model.
+    @property
+    def context_size(self) -> int:
+        """Context size of the model."""
+        return self.window * self.seasonality
 
-        Parameters
-        ----------
-        df:
-            Data to fit on
-        regressors:
-            List of the columns with regressors(ignored in this model)
+    def get_model(self) -> "SeasonalMovingAverageModel":
+        """Get internal model.
 
         Returns
         -------
         :
-            Fitted model
+           Internal model
         """
-        if set(df.columns) != {"timestamp", "target"}:
-            warnings.warn(
-                message=f"{type(self).__name__} does not work with any exogenous series or features. "
-                f"It uses only target series for predict/\n "
-            )
-
         return self
 
-    def forecast(self, df: pd.DataFrame, prediction_size: int) -> np.ndarray:
-        """Compute autoregressive forecasts.
+    def fit(self, ts: TSDataset) -> "SeasonalMovingAverageModel":
+        """Fit model.
+
+        For this model, fit does nothing.
 
         Parameters
         ----------
-        df:
-            Features dataframe.
-        prediction_size:
-            Number of last timestamps to leave after making prediction.
-            Previous timestamps will be used as a context for models that require it.
+        ts:
+            Dataset with features
 
         Returns
         -------
         :
-            Array with predictions.
-
-        Raises
-        ------
-        ValueError:
-            if context isn't big enough
-        ValueError:
-            if forecast context contains NaNs
+            Model after fit
         """
-        expected_length = prediction_size + self.shift
+        columns = set(ts.columns.get_level_values("feature"))
+        if columns != {"target"}:
+            warnings.warn(
+                message=f"{type(self).__name__} does not work with any exogenous series or features. "
+                f"It uses only target series for predict/\n "
+            )
+        return self
+
+    def _validate_prediction_size(self, df: pd.DataFrame, prediction_size: int):
+        expected_length = prediction_size + self.context_size
+
         if len(df) < expected_length:
             raise ValueError(
                 "Given context isn't big enough, try to decrease context_size, prediction_size of increase length of given dataframe!"
             )
 
-        history = df["target"][-expected_length:-prediction_size]
-        if np.any(history.isnull()):
+    def _forecast_df(self, df: pd.DataFrame, prediction_size: int) -> pd.DataFrame:
+        self._validate_prediction_size(df=df, prediction_size=prediction_size)
```

Comment:
Add docstring here

Reply:
For which method? `_forecast` or `_validate_prediction_size`?